### PR TITLE
feat(cli): enable using temp creds when initializing an env

### LINKED ISSUE #1115: Bug: copilot CLI fails if no default region found in profile
I am trying to set up copilot init demo app using the interactive CLI but I am unable as the default AWS profile does not have region specified (an optional field) so the init command fails.

```
hamidrasool@LAPTOP:~/copilot-demo-app$ copilot init
Note: It's best to run this command in the root of your Git repository.
Welcome to the Copilot CLI! We're going to walk you through some questions
to help you get set up with an application on ECS. An application is a collection of
containerized services that operate together.

Application name: copilot-demo
✘ get application copilot-demo: get application copilot-demo: MissingRegion: could not find region configuration
```

The missing feature in this case is a setting for --region parameter and optionally adding a prompt to the interactive init setup if it is not specified. Currently there is no such flag available, and surprised there isn't any planned in the roadmap.

```
hamidrasool@LAPTOP:~/copilot-demo-app$ copilot init --region eu-west-1
✘ unknown flag: --region
```

Current workaround is to add the region manually into `~/.aws/config` if needed to deploy for any particular region which should not be needed as it forces user to change their default settings for profile. Alternately setting a new profile for this purpose for each required region is equally not a pleasant user experience.

System tested on: Ubuntu 18.04 on WSL2

### LINKED ISSUE #1228: [design] Enable temporary creds for environment initialization
### Problem statement

As of Copilot v0.2.0, users cannot use temporary credentials while creating an environment (#1068, #1104).  
We want to support this use case, while still making it clear that the [default credential provider chain](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials) is used to look up application metadata and hence should prefer long lived profiles.

### Proposal

To satisfy the new requirement, we'll update the `env init` prompt as follows after the users input their environment name:
```bash
$ copilot env init

Name: prod-iad

These credentials will be used only for creating your environment.

Which credentials would you like to use to create prod-iad?
> Use temporary credentials
> [profile default]
> [profile test]
> [profile prod-iad]
> [profile prod-pdx]
```

If the user chooses a profile, then we'll continue down the existing flow and start creating the environment stack.

If user chooses to enter temporary credentials, then we will populate the defaults with values from their default credential provider. 

```bash
Name: prod-iad

These credentials will be used only for creating your environment.

Credential source: temporary credentials

  What's your AWS Access Key ID?     (***************ZDQY)
  What's your AWS Secret Access Key? (***************2pA/)
  What's your AWS Session Token ?    (****************asdf)
  Which region? (us-west-2)
```

We'll introduce new flags to match the new prompts:
```bash
$ copilot env init --help
Creates a new environment in your application.

Usage
  copilot env init [flags]

Flags
  -h, --help             help for init
  -n, --name string      Name of the environment.
      --prod             If the environment contains production services.
      --profile string   Name of the profile. Exclusive with temporary credential flags.  # New description.
      --aws-access-key-id string       Temporary access key ID.               # New flag.
      --aws-secret-access-key string   Temporary secret access key.           # New flag.
      --aws-session-token string       Temporary session token.               # New flag.
      --region string    Name of the region to create the environment in.     # New flag.

Global Flags
  -a, --app string   Name of the application.
```
The new `--region` flag allows us to satisfy https://github.com/aws/copilot-cli/issues/1115.

Finally, we will update `$ copilot init` to use the default credential provider instead of the "default" profile for initializing your "test" environment.

### PR DESCRIPTION
**Pull request by efekarakus**
Resolves #1228 and resolves #1115 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/term/selector/creds.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ func (s *CredsSelect) askTempCreds() (*session.Session, error) {
 	if err != nil {
 		return nil, fmt.Errorf("get secret access key: %w", err)
 	}
-	sessionToken, err := s.askWithMaskedDefault(sessionTokenPrompt, defaultSessToken)
+	sessionToken, err := s.askWithMaskedDefault(sessionTokenPrompt, defaultSessToken) // TODO(efekarakus): remove validation so that it can be empty.
```
Comment (efekarakus) [inline internal/pkg/term/selector/creds.go:80]: I'll tackle this change separately in parallel.

File: internal/pkg/cli/env_init.go [hunk 2]
```diff
@@ -297,32 +280,55 @@ func (o *initEnvOpts) askEnvName() error {
...
-	o.Profile = profile
+	o.sess.Config.Region = aws.String(region)
 	return nil
 }
 
 func (o *initEnvOpts) askCustomizedResources() error {
 	if o.NoCustomResources {
 		return nil
 	}
+	if o.selVPC == nil {
```
Comment (iamhopaul123) [inline internal/pkg/cli/env_init.go:328]: Is there any chance that this `o.selVPC` is not nil?

File: internal/pkg/cli/env_init.go [hunk 3]
```diff
@@ -220,17 +196,25 @@ func (o *initEnvOpts) Ask() error {
...
+	if err := o.askEnvRegion(); err != nil {
 		return err
 	}
 	return o.askCustomizedResources()
 }
 
 // Execute deploys a new environment with CloudFormation and adds it to SSM.
 func (o *initEnvOpts) Execute() error {
+	// Initialize environment clients if not set.
+	if o.envIdentity == nil {
```
Comment (iamhopaul123) [inline internal/pkg/cli/env_init.go:211]: same for `envIdentity` and `envDeployer`

File: internal/pkg/cli/env_init.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/cli/env_init.go:328]: The nil checks are for testing or the `init` command which sets these dependencies on its own

**Pull request merged**